src/common/reducers/snaps.js

```diff
@@ -31,6 +58,11 @@ export function snaps(state = {
         success: false,
         error: action.payload
       };
+    case RegisterNameActionTypes.REGISTER_NAME_SUCCESS:
+      return {
+        ...state,
+        snaps: updateRegisteredName(state.snaps, action.payload.id, action.payload.snapName)
```

Comment:
@katiefenn do you know any for redux based apps - because we are likely not getting rid of stores/reducers if we keep redux, but it would be useful to have some 'data structure' abstraction on top of it if anyone has done one already